src/figuresClasses.ts

```diff
@@ -1,19 +1,74 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
 export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Each side length must be greater than zero.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Wrong length sides of triangle');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
```

Comment:
you can destructure this
it will make you code shorter 
```suggestion
    const { a, b, c } = this;
    const s = (a + b + c) / 2;
```